pybuoy/mixins/parser.py

```diff
@@ -12,35 +14,32 @@
 from pybuoy.unit_mappings import MeteorologicalKey, WaveSummaryKey
 
 
-class XmlToDict(dict):
-    def __init__(self, data: Element):
-        if data.items():
-            self.update(dict(data.items()))
-        for element in data:
-            if element:
-                # treat like dict - assumes that if the first two tags
-                # in a series are different, then all are different.
-                if len(element) == 1 or element[0].tag != element[1].tag:
-                    tmp_dict: dict[str, XmlToDict] | XmlToDict = XmlToDict(element)
-                # treat like list - we assume that if the first two tags
-                # in a series are the same, then the rest are the same.
-                else:
-                    tmp_dict = {element[0].tag: XmlToDict(element)}
-                # if the tag has attributes, add those to the dict
-                if isinstance(element, XmlToDict) and element.items():
-                    tmp_dict.update(dict(element.items()))
-                self.update({element.tag: tmp_dict})
-            # assumes an attribute in a tag without any text.
-            elif element.items():
-                self.update({element.tag: dict(element.items())})
-            else:
-                self.update({element.tag: element.text})
-
-
-# TODO: refactor and incorporate mixin
 class ParserMixin:
     """Parser mixin supports handling of third-party data."""
 
+    def etree_to_dict(self, element: Element):
+        """Parse XML Element to dictionary."""
+        d: dict[str, dict[Any, Any] | Any] = {  # TODO: improve typing
+            element.tag: {} if element.attrib else None
```

Comment:
Why did you also adopt this pattern of one line conditionals? imo it makes things harder to understand but that's also because I'm not used to python syntax for one line conditionals.

Reply:
> Why did you also adopt this pattern of one line conditionals?
Only when logic is simple, I like to avoid verbosity of if-else blocks.

> ... I'm not used to python syntax for one line conditionals.
If you're more comfortable with JS, look up ternaries in JS; syntax is similar.

Reply:
> ... My suggestion is to have:
> 
> ```
> if element.attrib:
>   element.tag: {}
> else:
>   None
> ```
> 
> That's just my $0.02

FYI - that won't work as written; `element.tag` is a key that is set with either `{}` or `None` depending on whether `element.attrib` is truthy.